File: problem set 6/mario.py
from cs50 import get_int


def main():
  size = get_height()
  for row in range(size):
    for spaces in range((size - row - 1), 0, -1):
      print(" ", end="")
    for i in range(row + 1):
      print("#", end="")
    print()

def get_height():
  while True:
      try:
          height = get_int("Select a height from 1 to 8: ")
          if (height >= 1) and (height <= 8):
              break
      except:
          print("Please select a valid height from 1 to 8")
  return height


main()

File: cs50 Intro/problem set 1/mario.c
#include <cs50.h>
#include <stdio.h>

int get_size(void);
void print_grid(int size);

int main(void)
{
  // Get size of grid
  int n = get_size();

  // Print grid of bricks
  print_grid(n);
}

int get_size(void)
{
  int height;
  do
  {
    height = get_int("Size (between 1 and 8): ");
  } while (height < 1 || height > 8);
  return height;
}

void print_grid(int size)
{
  // Print each row up to the size
  for (int row = 0; row < size; row++)
  {
    // Print spaces starting from the size - 1 and working down
    for (int spaces = size - 1; spaces > row; spaces--)
    {
      printf(" ");
    }
    // Print blocks from 1 going up to the size
    for (int hashes = 0; hashes <= row; hashes++)
    {
      printf("#");
    }
    printf("\n");
  }
}